src/components/SearchBox/SearchBox.tsx

```diff
@@ -1,75 +1,75 @@
 /** @jsx h */
 
-import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
+import { h, createRef, Component } from 'preact';
 import { noop } from '../../lib/utils';
 import Template from '../Template/Template';
-
-const SearchBoxCSSClasses = PropTypes.shape({
-  root: PropTypes.string.isRequired,
-  form: PropTypes.string.isRequired,
-  input: PropTypes.string.isRequired,
-  submit: PropTypes.string.isRequired,
-  submitIcon: PropTypes.string.isRequired,
-  reset: PropTypes.string.isRequired,
-  resetIcon: PropTypes.string.isRequired,
-  loadingIndicator: PropTypes.string.isRequired,
-  loadingIcon: PropTypes.string.isRequired,
+import {
+  SearchBoxRendererCSSClasses,
+  SearchBoxTemplates,
+} from '../../widgets/search-box/search-box';
+
+type SearchBoxProps = {
+  placeholder?: string;
+  cssClasses: SearchBoxRendererCSSClasses;
+  templates?: SearchBoxTemplates;
+  query?: string;
+  showSubmit?: boolean;
+  showReset?: boolean;
+  showLoadingIndicator?: boolean;
+  refine?: (value: string) => void;
+  autofocus?: boolean;
+  searchAsYouType?: boolean;
+  isSearchStalled?: boolean;
+  disabled?: boolean;
+  onChange?: (event: Event) => void;
+  onSubmit?: (event: Event) => void;
+  onReset?: (event: Event) => void;
+} & typeof defaultProps;
+
+const defaultProps = Object.freeze({
+  query: '',
+  showSubmit: true,
+  showReset: true,
+  showLoadingIndicator: true,
+  autofocus: false,
+  searchAsYouType: true,
+  isSearchStalled: false,
+  disabled: false,
+  onChange: noop,
+  onSubmit: noop,
+  onReset: noop,
+  refine: noop,
 });
 
-class SearchBox extends Component {
-  static propTypes = {
-    placeholder: PropTypes.string.isRequired,
-    cssClasses: SearchBoxCSSClasses.isRequired,
-    templates: PropTypes.object.isRequired,
-    query: PropTypes.string,
-    showSubmit: PropTypes.bool,
-    showReset: PropTypes.bool,
-    showLoadingIndicator: PropTypes.bool,
-    refine: PropTypes.func,
-    autofocus: PropTypes.bool,
-    searchAsYouType: PropTypes.bool,
-    isSearchStalled: PropTypes.bool,
-    disabled: PropTypes.bool,
-    onChange: PropTypes.func,
-    onSubmit: PropTypes.func,
-    onReset: PropTypes.func,
-  };
+type SearchBoxState = {
+  query: string;
+  focused: boolean;
+};
 
-  static defaultProps = {
-    query: '',
-    showSubmit: true,
-    showReset: true,
-    showLoadingIndicator: true,
-    autofocus: false,
-    searchAsYouType: true,
-    isSearchStalled: false,
-    disabled: false,
-    onChange: noop,
-    onSubmit: noop,
-    onReset: noop,
-    refine: noop,
-  };
+class SearchBox extends Component<SearchBoxProps, SearchBoxState> {
+  public static defaultProps = defaultProps;
 
-  state = {
+  public state = {
```

Comment:
Can the state be made private here? I don't think we acccess it anywhere outside this class, do we?

Reply:
nevermind, it's public in Component<P,S>, game over :)